service/src/registry.ts

```diff
@@ -556,6 +638,443 @@ export class Registry {
+    if (!configuration.id) throw new ArgumentError('configuration object is missing id property');
+    /*Codes_SRS_NODE_IOTHUB_REGISTRY_18_009: [The `addConfiguration` method shall set `configuration.schemaVersion` to the constant `currentConfigurationSchemaVersion` if it is not already set. ]*/
+    if (!configuration.schemaVersion) {
+      configuration.schemaVersion = currentConfigurationSchemaVersion;
+    }
+
+    /*Codes_SRS_NODE_IOTHUB_REGISTRY_18_010: [The `addConfiguration` method shall construct an HTTP request using information supplied by the caller, as follows:
+    ```
+    PUT /configurations/<encodeURIComponent(configuration.id)>?api-version=<version> HTTP/1.1
+    Authorization: <sharedAccessSignature>
+    Content-Type: application/json; charset=utf-8
+    Request-Id: <guid>
+
+    <configuration>
+    ```
+    ]*/
+    const path = `/configurations/${encodeURIComponent(configuration.id)}${endpoint.versionQueryString()}`;
+    const httpHeaders = {
+      'Content-Type': 'application/json; charset=utf-8'
+    };
+
+    this._restApiClient.executeApiCall('PUT', path, httpHeaders, configuration, done);
+  }
+
+  /**
+   * @method            module:azure-iothub.Registry#getConfiguration
+   * @description       Get a single configuration from an IoT Hub
+   *
+   * @param {string}    configurationId   The ID of the configuration you with to retrieve
+   * @param {Function}  done              The callback which will be called with either an Error object
+   *                                      or a module:azure-iothub.Configuration object with the configuration details.
+   *
+   * @throws {ReferenceError}             The configurationId or done argument is falsy
+   */
+  getConfiguration(configurationId: string, done: (err: Error, configuration?: Configuration, response?: any) => void): void {
+    /*Codes_SRS_NODE_IOTHUB_REGISTRY_18_011: [The `getConfiguration` method shall throw a `ReferenceError` exception if `configurationId` is falsy. ]*/
+    if (!configurationId) throw new ReferenceError('Argument \'configurationId\' cannot be falsy');
+    if (!done) throw new ReferenceError('Argument \'done\' cannot be falsy');
+
+    /*Codes_SRS_NODE_IOTHUB_REGISTRY_18_012: [The `getConfiguration` method shall construct an HTTP request using information supplied by the caller, as follows:
+    ```
+    GET /configurations/<encodeURIComponent(configurationId)>?api-version=<version> HTTP/1.1
+    Authorization: <sharedAccessSignature>
+    Request-Id: <guid>
+    ```
+    ]*/
+    const path = `/configurations/${encodeURIComponent(configurationId)}${endpoint.versionQueryString()}`;
+    this._restApiClient.executeApiCall('GET', path, null, null, done);
+  }
+
+  /**
+   * @method            module:azure-iothub.Registry#getConfigurations
+   * @description       Get all configurations on an IoT Hub
```

Comment:
>Get all configurations on an IoT Hub [](start = 24, length = 36)

redundant comment but please double-check with the service team 1/ what's the max size 2/ if it should be a query